crates/oxc_linter/src/rules/eslint/no_implicit_coercion.rs

```diff
@@ -0,0 +1,593 @@
+                // Check for `${foo}` (string coercion via template literal)
+                // Skip if this is a tagged template literal (e.g., tag`${foo}`)
+                if self.string
+                    && self.disallow_template_shorthand
+                    && template.quasis.len() == 2
+                    && template.expressions.len() == 1
+                    && !is_tagged_template(ctx, node)
+                {
+                    let first_quasi = &template.quasis[0];
+                    let last_quasi = &template.quasis[1];
+
+                    // Check if both quasi parts contain no meaningful content.
+                    // We use the cooked value to handle escape sequences properly.
+                    // ESLint considers templates like `\n${foo}` as coercion because
+                    // escape sequences that become whitespace don't add meaningful content.
+                    // We also allow backslash characters in the cooked value because they
+                    // typically come from line continuations (e.g., `\\\n${foo}`) which
+                    // ESLint also treats as implicit coercion.
+                    let first_is_empty_or_whitespace = first_quasi
+                        .value
+                        .cooked
+                        .as_ref()
+                        .is_some_and(|s| s.chars().all(|c| c.is_whitespace() || c == '\\'));
+                    let last_is_empty_or_whitespace = last_quasi
+                        .value
+                        .cooked
+                        .as_ref()
+                        .is_some_and(|s| s.chars().all(|c| c.is_whitespace() || c == '\\'));
+
+                    if first_is_empty_or_whitespace && last_is_empty_or_whitespace {
+                        let expr = &template.expressions[0];
+                        // Don't warn if the expression is already a string literal or String() call
+                        if !is_string_literal_or_string_call(expr) {
+                            let operand = get_operand_text(ctx, expr);
+                            report_coercion(ctx, template.span(), CoercionKind::String, operand);
+                        }
+                    }
+                }
+            }
+            _ => {}
+        }
+    }
+}
+
+impl NoImplicitCoercion {
+    fn is_allowed(&self, operator: &str) -> bool {
+        self.allow.iter().any(|s| s.as_str() == operator)
+    }
+}
+
+fn get_operand_text<'a>(ctx: &LintContext<'a>, expr: &Expression<'a>) -> &'a str {
+    ctx.source_range(expr.without_parentheses().span())
+}
+
+fn is_numeric_literal(expr: &Expression) -> bool {
+    matches!(
+        expr.without_parentheses(),
+        Expression::NumericLiteral(_) | Expression::BigIntLiteral(_)
+    )
+}
+
+fn is_number_one(expr: &Expression) -> bool {
+    match expr.without_parentheses() {
+        Expression::NumericLiteral(lit) => (lit.value - 1.0).abs() < f64::EPSILON,
+        _ => false,
+    }
+}
```

Comment:
you can use `is_number_value` rather than this helper fn;
https://github.com/Boshen/oxc/blob/669afe058011b2c356f16c71c0a4b65dd181c973/crates/oxc_ast/src/ast_impl/js.rs#L137-L140